### PR TITLE
#339 expand text tool

### PR DESCRIPTION
**Pull request by irinatelnova**

### CODE REVIEW

File: packages/ketcher-react/src/script/ui/views/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ interface DialogProps {
   params: DialogParams
   buttons?: Array<string | React.ReactElement>
   className: string
+  manageFocus?: boolean
```
Comment (MagomedGasanov) [inline packages/ketcher-react/src/script/ui/views/components/Dialog/Dialog.tsx:28]: I think name 'manageFocus' is more appropriate for the function than for boolean type

**Review by AndreiMazol — CHANGES_REQUESTED** [review]
You must not use UI specific library in core package

File: packages/ketcher-core/src/chemistry/entities/Text.ts [hunk 2]
```diff
@@ -15,21 +15,21 @@
  ***************************************************************************/
 
 import { Vec2 } from 'utils'
+import { RawDraftContentState } from 'draft-js'
```
Comment (AndreiMazol) [inline packages/ketcher-core/src/chemistry/entities/Text.ts:18]: Please remove 'draft-js' dependency from core package

File: packages/ketcher-core/src/chemistry/entities/Text.ts [hunk 3]
```diff
@@ -15,21 +15,21 @@
...
 
 export interface TextParams {
-  label: string
+  rawContent: RawDraftContentState
```
Comment (AndreiMazol) [inline packages/ketcher-core/src/chemistry/entities/Text.ts:21]: Please rename it to label or content

File: packages/ketcher-core/src/chemistry/entities/Text.ts [hunk 4]
```diff
@@ -15,21 +15,21 @@
...
   position?: Vec2
   type: string
 }
 
 export class Text {
-  label?: string | null
-  position?: Vec2
+  rawContent: RawDraftContentState
+  position: Vec2
   type: string
```
Comment (AndreiMazol) [inline packages/ketcher-core/src/chemistry/entities/Text.ts:29]: What do you need 'type' property for?

File: packages/ketcher-react/src/script/editor/tool/text.ts [hunk 5]
```diff
@@ -17,10 +17,15 @@
...
-  fromTextDeletion
+  fromTextDeletion,
+  TextElemData
 } from '../actions/text'
 import { fromMultipleMove } from '../actions/fragment'
 import Action from '../shared/action'
+import Editor from '../Editor'
+import { RawDraftContentState } from 'draft-js'
+import { Vec2 } from 'ketcher-core'
+import _ from 'lodash'
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/editor/tool/text.ts:28]: Please import single lodash function instead

File: packages/ketcher-react/src/script/render/restruct/ReText.ts [hunk 6]
```diff
@@ -15,14 +15,22 @@
...
+import { scale, Vec2, Text } from 'ketcher-core'
 import { LayerMap } from './GeneralEnumTypes'
+import {
+  DraftInlineStyleType,
+  RawDraftContentBlock,
+  RawDraftInlineStyleRange
+} from 'draft-js'
+import _ from 'lodash'
+import ReStruct from './ReStruct'
+import { TextStyle } from '../../ui/views/modal/components/Text/components/TextControlPanel'
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/render/restruct/ReText.ts:27]: It looks strange

File: packages/ketcher-react/src/script/render/restruct/ReText.ts [hunk 7]
```diff
@@ -50,55 +58,177 @@ class ReText extends ReObject {
...
-    this.item.label = this.item.label.replace(/[^\S\r\n]/g, '\u00a0')
 
-    this.path = paper.text(paperScale.x, paperScale.y, this.item.label).attr({
-      font: options.font,
-      'font-size': options.fontsz,
-      'text-anchor': 'start',
-      fill: '#000000'
+    let shiftY: number = 0
+    this.paths = []
+    this.item.rawContent.blocks.forEach((block: RawDraftContentBlock) => {
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/render/restruct/ReText.ts:123]: Potentially exception can be thrown here

File: packages/ketcher-react/src/script/render/restruct/ReText.ts [hunk 8]
```diff
@@ -15,14 +15,22 @@
...
 import ReObject from './ReObject'
-import { Box2Abs, scale, Vec2 } from 'ketcher-core'
+import { scale, Vec2, Text } from 'ketcher-core'
 import { LayerMap } from './GeneralEnumTypes'
+import {
+  DraftInlineStyleType,
+  RawDraftContentBlock,
+  RawDraftInlineStyleRange
+} from 'draft-js'
+import _ from 'lodash'
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/render/restruct/ReText.ts:25]: Please import single lodash function instead

File: packages/ketcher-react/src/script/render/restruct/ReText.ts [hunk 9]
```diff
@@ -50,55 +58,177 @@ class ReText extends ReObject {
...
       LayerMap.data,
       this.visel,
-      this.path,
+      _.flatten(this.paths),
       null,
       true
     )
   }
+
+  getRanges(
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/render/restruct/ReText.ts:168]: Please follow SOLID principles. This object has lots of responsibilities

File: packages/ketcher-react/src/script/ui/views/components/Dialog/Dialog.tsx [hunk 10]
```diff
@@ -51,11 +52,14 @@ const Dialog: FC<Props> = props => {
     valid = () => !!result(),
     buttons = ['Cancel', 'OK'],
     className,
+    manageFocus,
     ...rest
   } = props
   const dialogRef = useRef<HTMLDivElement>(null)
 
   useEffect(() => {
+    if (manageFocus === false) return
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/ui/views/components/Dialog/Dialog.tsx:61]: It looks strange

**Review by AndreiMazol — CHANGES_REQUESTED** [review]

**Pull request closed without merging**